### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by jacksonargo**
Possibly the laziest makefile ever written

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -18,49 +18,11 @@ jobs:
...
-          for tag in $(docker run --rm artifact --release-tags); do
-              docker tag builder $IMAGE_ID:$tag-builder
-              docker tag artifact $IMAGE_ID:$tag
-              docker tag object-cache $OBJECT_CACHE_IMAGE_ID:$tag
-              docker tag page-cache $PAGE_CACHE_IMAGE_ID:$tag
-              docker push $IMAGE_ID:$tag-builder
-              docker push $IMAGE_ID:$tag
-              docker push $OBJECT_CACHE_IMAGE_ID:$tag
-              docker push $PAGE_CACHE_IMAGE_ID:$tag
+          for tag in latest $(docker run --rm artifact --release-tags); do
```
Comment (jacksonargo) [inline .github/workflows/build.yml:26]: Should release tags be a target :thinking: Maybe... not sure.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+VVGO_IMAGE_NAME=vvgo
+PAGE_CACHE_IMAGE_NAME=page-cache
+OBJECT_CACHE_IMAGE_NAME=object-cache
+
+GITHUB_REPO ?= virtual-vgo/vvgo
+IMAGE_REPO ?= docker.pkg.github.com/$(GITHUB_REPO)
+VVGO_IMAGE_CACHE ?= $(IMAGE_REPO)/$(VVGO_IMAGE_NAME):latest
+PAGE_CACHE_IMAGE_CACHE ?= $(IMAGE_REPO)/$(PAGE_CACHE_IMAGE_NAME):latest
+OBJECT_CACHE_IMAGE_CACHE ?= $(IMAGE_REPO)/$(OBJECT_CACHE_IMAGE_NAME):latest
```
Comment (0az) [inline Makefile:37]:
```suggestion
REGISTRY ?= docker.pkg.github.com/$(GITHUB_REPO)
VVGO_IMAGE_CACHE ?= $(REGISTRY)/$(VVGO_IMAGE_NAME):latest
PAGE_CACHE_IMAGE_CACHE ?= $(REGISTRY)/$(PAGE_CACHE_IMAGE_NAME):latest
OBJECT_CACHE_IMAGE_CACHE ?= $(REGISTRY)/$(OBJECT_CACHE_IMAGE_NAME):latest
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+images/vvgo: images/vvgo-builder
+	docker pull $(VVGO_IMAGE_CACHE) || true
+	docker build . \
+		--file Dockerfile \
+		--target vvgo \
+		--cache-from=$(VVGO_IMAGE_CACHE)-builder \
+		--cache-from=$(VVGO_IMAGE_CACHE) \
+		--tag artifact
+
+images/page-cache:
```
Comment (0az) [inline Makefile:60]: IMHO we should really rename this to ingress or (reverse?) proxy. Maybe later, though.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	docker build infra/page-cache \
+		--file infra/page-cache/Dockerfile \
+		--cache-from=$(OBJECT_CACHE_IMAGE_CACHE) \
+		--tag page-cache
+
+images: images/vvgo images/page-cache images/object-cache
+
+# Push images
+
+RELEASE_TAG ?= development
```
Comment (0az) [inline Makefile:78]: Why not add a hash?

File: .github/workflows/build.yml (hunk 1 above)
Comment (0az) [inline .github/workflows/build.yml:26]: I'm still not a fan of the `--release-tags` flag: should probably get extracted.

**Pull request merged**